spec/models/tasks/cherrypick_task_spec.rb

```diff
@@ -1,6 +1,150 @@
 require 'rails_helper'
 
 RSpec.describe CherrypickTask, type: :model do
+  let!(:plate) { create :plate_with_untagged_wells, sample_count: 4 }
+  let!(:control_plate) { create :plate_with_untagged_wells, sample_count: 2 }
```

Comment:
You should be able to use that 'control_plate' factory that we made with 2 samples to to give you what you need instead of this 'plate_with_untagged_wells' which will not be the right plate class (ControlPlate). Can then remove the control sample updates from the before block below.

Reply:
Ok, changed